ludwig/automl/automl.py

```diff
@@ -171,28 +187,91 @@ def train_with_config(
     return AutoTrainResults(experiment_analysis)
 
 
-def _model_select(default_configs):
+def _model_select(
+    dataset: Union[str, pd.DataFrame, dd.core.DataFrame, DatasetInfo],
+    default_configs,
+    user_specified_config,
+):
     """
-    Performs model selection based on dataset.
-    Note: Current implementation returns tabnet by default. This will be
-        improved in subsequent iterations
+    Performs model selection based on dataset or user specified model.
+    Note: Current implementation returns tabnet by default. If the
+    percentage of numerical features is >90%, the concat model is used.
     """
-    return default_configs['concat']
+
+    dataset_info = get_dataset_info(dataset) if not isinstance(
+        dataset, DatasetInfo) else dataset
+    fields = dataset_info.fields
+    row_count = dataset_info.row_count
+
+    total_numerical_feats = 0
+
+    for idx, field in enumerate(fields):
+        missing_value_percent = 1 - float(field.nonnull_values) / row_count
+        dtype = infer_type(field, missing_value_percent)
+        if dtype == NUMERICAL:
+            total_numerical_feats += 1
+
+    percent_numerical_feats = total_numerical_feats / len(fields)
+
+    base_config = default_configs["base_config"]
+
+    # tabular dataset heuristics
+    if len(fields) > 3:
+        if percent_numerical_feats > 0.9:
+            base_config = (
```

Comment:
This line seems to be missing a merge_dict, so when concat is selected, bad json config is created.  Saw this when running a version of this change set on tf-legacy on the higgs dataset.